src/utils/config.ts

```diff
@@ -186,10 +200,26 @@ export class ConfigClient {
       signer ?? providerUtils.getProvider(this.getHubPoolChainId());
     return AcceleratingDistributor__factory.connect(address, provider);
   }
-  getMerkleDistributor(signer?: Signer): AcrossMerkleDistributor {
-    const address = this.getMerkleDistributorAddress();
-    const provider =
-      signer ?? providerUtils.getProvider(this.getHubPoolChainId());
+  getMerkleDistributor(
+    rewardsType: constants.rewardProgramTypes,
+    signer?: Signer
+  ): AcrossMerkleDistributor {
+    let address = "";
+
+    if (rewardsType === "referrals") {
+      address = this.getMerkleDistributorAddress();
+    } else if (rewardsType === "op-rebates") {
+      address = this.getOpRewardsMerkleDistributorAddress();
+    }
```

Comment:
nit: thought on this:
```suggestion
    let address = rewardsType === "referrals" ?
        this.getMerkleDistributorAddress() : 
         this.getOpRewardsMerkleDistributorAddress();
```